### PR TITLE
Fix building on Solaris with data_packaging=library

### PR DESCRIPTION
**Pull request by datalogics-kam**
The problem with the build was that the library directories were added to the incorrect library path. The code uses `LD_LIBRARY_PATH` by default, but on our Solaris machines, we want `LD_LIBRARY_PATH_64`.

There is a Make variable `LDLIBRARYPATH_ENVVAR` that controls the variable name. Set that variable on Solaris 64-bit to use `LD_LIBRARY_PATH_64`.

In the process, generalized the way arguments are presented to the make command.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ class ICUConan(ICUBase):
                        "with_extras": True,
                        "with_unit_tests": False,
                        "silent": True}
+    generators = 'txt', 'virtualenv'
```
Comment (datalogics-robb) [inline conanfile.py]: This looks like it might have just snuck in - is this addition required for fixing the Solaris issue?

File: icu_base.py [hunk 2]
```diff
@@ -117,16 +121,17 @@ def build(self):
 
                     self.run(self._build_config_cmd, win_bash=tools.os_info.is_windows)
                     if self.options.get_safe("silent"):
```
Comment (datalogics-robb) [inline icu_base.py]: original code seems like it's double checking - can get_safe() return something that evaluates as a boolean different from ```self.options.slient```?  I would think not...

File: conanfile.py (hunk 1 above)
Comment (datalogics-kam) [reply conanfile.py]: It was required for diagnosis, and I left it in. I mean, I can take it out, but it's manifestly useful to be able to activate the build environment when manually developing a Conan package.

File: icu_base.py (hunk 2 above)
Comment (datalogics-kam) [reply icu_base.py]: `get_safe()` returns the value or `None` if it's not defined. `None` has falsy value.

The problem with the old code is that if `silent` was `False`, it would use the `else:` clause and set `--silent` anyway. Not helpful for debugging.

`icu_base.py` is subclassed by one build that has a `silent` option, and one that does not, which is why it's using `get_safe()`
